torchtune/training/quantization.py

```diff
@@ -74,6 +82,45 @@ def quantize(self, model):
 ] = enable_8da4w_fake_quant_module_swap
 
 
+class Int8MixedPrecisionTrainingQuantizer:
+    """Apply INT8 mixed-precision training. During training, weights and activations
+    are dynamically quantized to INT8 to utilize INT8 tensor cores. This is also done
+    in the backward pass.
+
+    NOTE: due to the limitations of the current implementation, the following
+    requirements must be satisfied to enjoy speedup:
+
+    1. Must use ``torch.compile()`` (set ``compile=True``).
+    2. Inputs to the model must not be too dynamic e.g. input sequence length changes
+    for every batch.
+
+    To satisfy (2), you can use :class:`~torchtune.datasets.PackedDataset` (set
+    ``dataset.packed=True``), which ensures input tokens always have fixed length.
```

Comment:
I believe that the user also has to set tokenizer.max_seq_len. @RdoubleA , can you confirm?

Reply:
Yes, `tokenizer.max_seq_len` must be set too. An error will be raised and ask to set `tokenizer.max_seq_len` when it is not set while `dataset.packed=True` is set. I will add it for completeness.